src/librustc/mir/interpret/mod.rs

```diff
@@ -1,8 +1,66 @@
 //! An interpreter for MIR used in CTFE and by miri
 
 #[macro_export]
-macro_rules! err {
-    ($($tt:tt)*) => { Err($crate::mir::interpret::InterpError::$($tt)*.into()) };
+macro_rules! throw_err_unsup {
+    ($($tt:tt)*) => {
+        Err($crate::mir::interpret::InterpError::Unsupported(
```

Comment:
My proposal was for `throw_` to do the `return` as well. Otherwise it's not really "throwing".

Reply:
Also I think `throw_X` can actually make good use of the `err_X` macro.

Reply:
Oh also `throw_err` seems somewhat redundant. I had [explicitly suggested](https://github.com/rust-lang/rust/pull/62969#discussion_r308574144) `throw_ub` and the like, now I wonder if there is a good reason for your more verbose version?

Reply:
I love how particular we are about stuff :) Will fix these right away.

Reply:
And apologies for the back and forth!

Reply:
It's okay, this is part of the mentoring service. ;) I just sometimes am worried that I annoy people by continuously finding new problems in their code... it can be hard to find the right balance. But after all, once this code lands, we all have to collectively maintain it.

So, thanks for your endurance!

Reply:
I've made the changes :)

Reply:
Thanks, looking good. :)